services/spd/trusty/trusty.c

```diff
@@ -398,17 +410,29 @@ static int32_t trusty_setup(void)
 		return -1;
 	}
 
-	/* Trusty runs in AARCH64 mode */
-	SET_PARAM_HEAD(ep_info, PARAM_EP, VERSION_1, SECURE | EP_ST_ENABLE);
-	ep_info->spsr = SPSR_64(MODE_EL1, MODE_SP_ELX, DISABLE_ALL_EXCEPTIONS);
+	instr = *(uint32_t *)ep_info->pc;
 
-	/*
-	 * arg0 = TZDRAM aperture available for BL32
-	 * arg1 = BL32 boot params
-	 * arg2 = BL32 boot params length
-	 */
-	ep_info->args.arg1 = ep_info->args.arg2;
```

Comment:
Are these parameters no longer required by Trusty?

Reply:
These are platform specific. If you need them on your platform, you can set them in plat_trusty_set_boot_args.

Reply:
But now you removed the previous code. Shouldnt we be making changes in such a way that current platforms do not break?

Reply:
Can you add the hooks to your platform code? This patch restores the part of the original trusty patches we need to boot on other platforms. Those platforms stopped working after syncing your changes.  I don't know why you decided to remove the mechanism we had for platform specific initialization and add your init code to the generic trusty code instead. Since I don't know which platform(s) rely on this, and since I cannot test them, I did not touch them.

Reply:
I added a patch to add a plat_trusty_set_boot_args hook to the tegra platform. Please test it as I don't have any hardware to test it on.